docs/source/export.rst

```diff
@@ -1,11 +1,293 @@
+    from torch.export import export, dynamic_dim
+
+    class M(torch.nn.Module):
+        def __init__(self):
+            super().__init__()
+
+            self.branch1 = torch.nn.Sequential(
+                torch.nn.Linear(64, 32), torch.nn.ReLU()
+            )
+            self.branch2 = torch.nn.Sequential(
+                torch.nn.Linear(128, 64), torch.nn.ReLU()
+            )
+
+        def forward(self, x1, x2):
+            out1 = self.branch1(x1)
+            out2 = self.branch2(x2)
+            return (out1, out2)
+
+    example_args = (torch.randn(32, 64), torch.randn(32, 128))
+    constraints = [
+        # First dimension of each input is a dynamic batch size
+        dynamic_dim(example_args[0], 0),
+        dynamic_dim(example_args[1], 0),
+        # The dynamic batch size between the inputs are equal
+        dynamic_dim(example_args[0], 0) == dynamic_dim(example_args[1], 0),
+    ]
+
+    exported_program: torch.export.ExportedProgram = export(
+      M(), args=example_args, constraints=constraints
+    )
+    print(exported_program)
+    """
+    ExportedProgram:
+        class GraphModule(torch.nn.Module):
+            def forward(self, arg0_1: f32[32, 64], arg1_1: f32[32], arg2_1: f32[64, 128], arg3_1: f32[64], arg4_1: f32[s0, 64], arg5_1: f32[s1, 128]):
+                # code: out1 = self.branch1(x1)
+                permute: f32[64, 32] = torch.ops.aten.permute.default(arg0_1, [1, 0]);  arg0_1 = None
+                addmm: f32[s0, 32] = torch.ops.aten.addmm.default(arg1_1, arg4_1, permute);  arg1_1 = arg4_1 = permute = None
+                relu: f32[s0, 32] = torch.ops.aten.relu.default(addmm);  addmm = None
+
+                # code: out2 = self.branch2(x2)
+                permute_1: f32[128, 64] = torch.ops.aten.permute.default(arg2_1, [1, 0]);  arg2_1 = None
+                addmm_1: f32[s1, 64] = torch.ops.aten.addmm.default(arg3_1, arg5_1, permute_1);  arg3_1 = arg5_1 = permute_1 = None
+                relu_1: f32[s1, 64] = torch.ops.aten.relu.default(addmm_1);  addmm_1 = None
+                return (relu, relu_1)
+
+    Graph signature: ExportGraphSignature(parameters=['L__self___branch1_0.weight', 'L__self___branch1_0.bias', 'L__self___branch2_0.weight', 'L__self___branch2_0.bias'], buffers=[], user_inputs=['arg4_1', 'arg5_1'], user_outputs=['relu', 'relu_1'], inputs_to_parameters={'arg0_1': 'L__self___branch1_0.weight', 'arg1_1': 'L__self___branch1_0.bias', 'arg2_1': 'L__self___branch2_0.weight', 'arg3_1': 'L__self___branch2_0.bias'}, inputs_to_buffers={}, buffers_to_mutate={}, backward_signature=None, assertion_dep_token=None)
+    Range constraints: {s0: RangeConstraint(min_val=2, max_val=9223372036854775806), s1: RangeConstraint(min_val=2, max_val=9223372036854775806)}
+    Equality constraints: [(InputDim(input_name='arg4_1', dim=0), InputDim(input_name='arg5_1', dim=0))]
+    """
+
+Some additional things to note:
+
+* Through the :func:``torch.export.dynamic_dim`` API, we specified the first
```

Comment:
Not rendering correctly
<img width="395" alt="image" src="https://github.com/pytorch/pytorch/assets/5652049/cbb41d1f-aca1-4a70-b49c-92a343cee396">